mercantile/__init__.py

```diff
@@ -191,8 +204,13 @@ def bounds(*tile):
     """
     tile = _parse_tile_arg(*tile)
     xtile, ytile, zoom = tile
+
+    # TODO: b gets constructed as potentially invalid Tile
+    # e.g. (1,1,0) is invalid, as is (1, 2, 1)
```

Comment:
Here is the problem: the like

    b = ul(xtile + 1, ytile + 1, zoom)

always constructs invalid tiles at the edges e.g. for (1, 0, 1) we create (2, 0, 1) which is not valid either

Reply:
Line 212 has been bugging me for a while. We know the dimensions of any tile in mercator units, it's a function of zoom only. So we can compute `b` without needing the concept of the neighboring tile to the lower right.